nav2_dwb_controller/dwb_core/src/publisher.cpp

```diff
@@ -148,7 +148,7 @@ DWBPublisher::publishEvaluation(std::shared_ptr<dwb_msgs::msg::LocalPlanEvaluati
 {
   if (results == nullptr) {return;}
 
-  if (publish_evaluation_) {
+  if (publish_evaluation_ && (node_->count_subscribers(eval_pub_->get_topic_name()) > 0)) {
```

Comment:
You don't need the surrounding parentheses,  `if (publish_evaluation_ && node_->count_subscribers(eval_pub_->get_topic_name()) > 0) {` would work fine :-)